### PR TITLE
fix oc explain status and spec verification

### PR DESCRIPTION
**Pull request by atiratree**
to prevent the test failures in oc kube bump: 

- https://github.com/openshift/oc/pull/1420
- https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1420/pull-ci-openshift-oc-master-e2e-aws-ovn/1660630230638792704

see aditional explanation in alternative PR by @ardaguclu : https://github.com/openshift/origin/pull/27936#issue-1723706437

### CODE REVIEW

File: test/extended/cli/explain.go
Comment (atiratree) [reply test/extended/cli/explain.go:623]: ye, it is just a quick solution to make it work for the types we test (`builtinTypes`). It will not work for the special cases like `ingresses`, etc. 

- We could fix this later once we add such a special type in the future. 
- Or I can add a prerequisite that a singular should be specified for each resource in the test in builtinTypes. 
- Or create a singularExceptions map for a few of these resources. Thoughts?

Comment (ardaguclu) [reply test/extended/cli/explain.go:623]: After thinking on this; maybe it is not necessary try to add singular name of resource between `<` and `>`. It seems to me that the maintenance effort outweighs the value we get comparing to the simply `<.*>`

Comment (atiratree) [reply test/extended/cli/explain.go:623]: Up to you, but I think it is safe to leave it since we are probably not expecting new builtin types. And if there are new in the future, exception can be made by the person introducing these.

Comment (ardaguclu) [reply test/extended/cli/explain.go:623]: I agree that it would be better being at safe side but I'd still prefer not stealing someone's at least 4 hours in the future by waiting for the CI and after failing, adding an exception into this regex :). Let's defer final decision to @soltysh as an approver.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
